src/main.c

```diff
@@ -0,0 +1,24 @@
+#include <stdint.h>
+
+#include "gpio.h"
+
+int main()
+{
+	volatile uint32_t *RCC_AHBENR = (uint32_t *)0x40021014;
+
+	*RCC_AHBENR |= (1 << 17);
+
+	GPIO_init_config PA5_config;
```

Comment:
nitpick: designated initializers would be nice here:

```suggestion
	GPIO_init_config PA5_config = {
	  .pin = GPIO_PIN_5,
	  .mode = GPIO_MODE_OUTPUT,
	  .pull = GPIO_NO_PUPD,
	  .speed = GPIO_SPEED_LOW,
	};
```